tests/test_endpoints.py

```diff
@@ -0,0 +1,79 @@
+import pytest
+import json
+from app import create_app
+import sys
+sys.path.append("..")
+
+
+@pytest.yield_fixture
+def app():
+    app = create_app("testing")
+    yield app
+
+
+@pytest.fixture
+def test_cli(loop, app, sanic_client):
+    return loop.run_until_complete(sanic_client(app))
+
+
+async def test_index(test_cli):
+    resp = await test_cli.get('/')
+    assert resp.status == 200
+
+
+async def test_getLanguages_with_no_query_param(test_cli):
+    resp = await test_cli.get('/languages')
+    resp_json = await resp.json()
+    assert resp_json["total_results"] == 0
+    assert resp_json["page"] == 1
+    assert resp.status == 200
+
+
+async def test_getLanguages_with_more_query_param(test_cli):
+    params = {'page': 2, 'type': 'local', 'limit': 50}
+    resp = await test_cli.get('/languages', data=json.dumps(params))
+    resp_json = await resp.json()
+    assert resp_json["message"] == "Expecting less than 3 arguments"
+    assert resp.status == 400
+
+
+async def test_getLanguages_with_valid_query_param(test_cli):
+    params = {'page': 2, 'type': 'local'}
+    resp = await test_cli.get('/languages', data=json.dumps(params))
+    resp_json = await resp.json()
+    assert resp_json["total_results"] == 0
+    assert resp_json["page"] == 2
+    assert resp.status == 200
+
+
+async def test_getLanguages_with_query_param_page(test_cli):
```

Comment:
What is the difference between this test and the test on top of this one? It looks identical if it was not for the fact that this does not take the type. In situations like this, you should delete this one and keep only the one above. Also, the name is suggesting some query parameter but in fact, that is still sent as body?

Reply:
I am trying to test the endpoint by simulating with all possible query parameters